tests/chainer_tests/functions_tests/activation_tests/test_softmax.py

```diff
@@ -101,6 +104,33 @@ def test_backward_gpu_non_contiguous(self):
     def test_backward_gpu_no_cudnn(self):
         self.check_backward(cuda.to_gpu(self.x), cuda.to_gpu(self.gy), 'never')
 
+    def check_double_backward(self, x_data, gy_data, ggx_data,
+                              use_cudnn='always'):
+        def f(x):
+            return functions.softmax(x, axis=self.axis)
+
+        with chainer.using_config('use_cudnn', use_cudnn):
+            gradient_check.check_double_backward(
+                f, (x_data,), (gy_data,), (ggx_data),
+                **self.check_backward_options)
```

Comment:
Sorry, one another comment: it seems this test is numerically unstable. I recommend to use a slightly looser threshold for double backprop test here.

Reply:
I fixed to give looser threshold by an order for double backprop test in case that dtype is float16. Is it enough or better to fix for other dtypes too?